python3Packages.schwifty: init at 2022.6.0

milibopp opened the pull request:
###### Description of changes

added python3Packages.schwifty

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] ~Tested, as applicable:~ (not applicable)
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] ~Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)~ (not applicable)
- [ ] ~Tested basic functionality of all binary files (usually in `./result/bin/`)~ (not applicable)
- [x] Tested import Python package and using basic features
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] ~(Package updates) Added a release notes entry if the change is major or breaking~ (not applicable)
  - [ ] `(Module updates) Added a release notes entry if the change is significant~ (not applicable)
  - [ ] ~(Module addition) Added a release notes entry if adding a new NixOS module~ (not applicable)
  - [ ] ~(Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes~ (not applicable)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

milibopp committed:
python3Packages.schwifty: init at 2022.6.0

fabaff commented on pkgs/development/python-modules/schwifty/default.nix:
```diff
@@ -0,0 +1,26 @@
+  pname = "schwifty";
+  version = "2022.6.0";
+  format = "pyproject";
+
```
No obsolete Python releases are [supported](https://github.com/mdomke/schwifty/blob/master/setup.cfg#L34).

```suggestion

disabled = pythonOlder "3.7";
```

fabaff commented on pkgs/development/python-modules/schwifty/default.nix:
```diff
@@ -0,0 +1,26 @@
+  };
+
+  propagatedBuildInputs = [ iso3166 pycountry ];
+
```
The project seems to be using `pytest` to run the tests. `pythonImportsCheck` is an additional safe-guard.

```suggestion

  checkInputs = [
    pytestCheckHook
  ];

  pythonImportsCheck = [
    "schwifty"
  ];
```

fabaff replied:
It could also be required that you patch out the coverage part of the tests. `tox.ini` is mentioning `pytest-cov`.

milibopp commented:
Thanks for your feedback, @fabaff. I'm looking into it.

milibopp commented:
Included comments by @fabaff, fixed build and rebased on master again.

I decided to just include pytest-cov under `checkInputs` instead of patching its usage, as feels easier to maintain this way. Hope that makes sense.

fabaff commented:
Result of `nixpkgs-review pr 177025` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.schwifty</li>
    <li>python39Packages.schwifty</li>
  </ul>
</details>

fabaff commented:
> I decided to just include pytest-cov under checkInputs instead of patching its usage, as feels easier to maintain this way. Hope that makes sense.

The coverage doesn't add value for us. A lot of the Python people here remove everything related to the coverage but ultimately its the maintainer's call.

fabaff reviewed (approved):
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/177025#issuecomment-1158175432)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-177025/results/python310Packages.schwifty/lib/python3.10/site-packages]$ python
Python 3.10.4 (main, Mar 23 2022, 20:25:24) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import schwifty
>>> iban = schwifty.IBAN.generate('DE', bank_code='10010010', account_code='12345')
>>> iban.checksum_digits
'40'
```

Looks good to me.
